### ISSUE TITLE
Panic in kube-utils after config map update error

### ISSUE THREAD

**Issue by scritchley** [weaveworks/weave#3613]
## What you expected to happen?
kube-utils should gracefully handle errors when updating config map annotations.

## What happened?
We are seeing a panic of kube-utils in our weave pods shortly after any autoscaling events / node terminations. The panic follows this error:
```
ERRO: 2019/03/11 00:00:30.580196 [kube-peers] error from UpdateAnnotation:  Operation cannot be fulfilled on configmaps "weave-net": the object has been modified; please apply your changes to the latest version and try again

```

The full panic output is attached:
[kube-utils-panic.txt](https://github.com/weaveworks/weave/files/2953052/kube-utils-panic.txt)

The panic occurs when writing to the annotations map of the ConfigMap here:
https://github.com/weaveworks/weave/blob/master/prog/kube-utils/annotations.go#L104

The code assumes the Annotations map will always be non-nil, however, when an error is returned 
from `(*configMaps).Update` method, the k8s client returns a new zero valued *v1.ConfigMap, see:
https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/configmap.go#L123

This results in subsequent calls to UpdateAnnotation panicking due to a nil Annotations map. This results in the kube-utils -run-reclaim-daemon process exiting and the weave pods continuously attempting to connect to dead peers.

## How to reproduce it?
To replicate the issue, an error needs to be returned from the UpdateAnnotation method. In practice, our clusters (~50 nodes each) often return errors due to (what I assume are) contentious updates to the config map. The panic occurs shortly thereafter.

## Anything else we need to know?
- Deployed on AWS using Kops
- 3 clusters of ~50 nodes each regularly experiencing this panic

## Versions:

Weave version: 2.5.0
Docker version: 18.06.1
OS: Linux 4.9.0-7-amd64 SMP Debian 4.9.110-3+deb9u2 (2018-08-13) x86_64 Linux
Kubectl: 
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-01T20:08:12Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.6", GitCommit:"b1d75deca493a24a2f87eb1efde1a569e52fc8d9", GitTreeState:"clean", BuildDate:"2018-12-16T04:30:10Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by syscll** [comment]
Hi, @bboreham. As mentioned in the linked PR, I'm more than happy to help fix this.

We've been running the PR version in 4 clusters of 50+ nodes for over a week, but I anticipate you might have a different solution for the fix.

**Comment by bboreham** [comment]
Closed by #3623
